sdcm/utils/issues.py

```diff
@@ -0,0 +1,93 @@
+    issue_id: int | None
+
+
+class SkipPerIssues:
+    def __init__(self, issues: list[str] | str, params: SCTConfiguration | dict):
+        self.params = params
+
+        github_access = KeyStore().get_json("github_access.json")
+        auth = github.Auth.Token(token=github_access["token"])
+        self.github = github.Github(auth=auth)
+
+        issues = [issues] if isinstance(issues, str) else issues
+
+        self.issues = [self.get_issue_details(issue) for issue in issues]
+        self.issues = list(filter(lambda x:  x, self.issues))  # filter None - unmatched issues
+
+    @staticmethod
+    def parse_issue(issue_str):
+        for pattern in [scylla_issue_pattern, scylla_issue_https_pattern]:
+            match = pattern.search(issue_str.strip())
+            if match:
+                obj = match.groupdict()
+                return Issue(
+                    user_id=obj.get("user_id") or "scylladb",
+                    repo_id=obj.get("repo_id") or "scylladb",
+                    issue_id=int(obj.get("issue_id"))
+                )
+        return None
+
+    @lru_cache
+    def get_issue_details(self, issue):
+        issue_parsed = self.parse_issue(issue)
+        if not issue_parsed:
+            logging.warning("couldn't parse issue: %s", issue)
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"couldn't parse issue: {issue}",
+                               severity=Severity.WARNING,
+                               trace=sys._getframe().f_back).publish()  # pylint: disable=protected-access
+            return None
+        try:
+            return self.github.get_user(issue_parsed.user_id).get_repo(issue_parsed.repo_id).get_issue(issue_parsed.issue_id)
+        except UnknownObjectException:
+            logging.warning("couldn't find issue: %s", issue)
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"couldn't find issue: {issue}",
+                               severity=Severity.WARNING,
+                               trace=sys._getframe().f_back).publish()  # pylint: disable=protected-access
+            return None
+        except Exception as exc:
+            logging.warning("failed to get issue: %s", issue)
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"failed to get issue {issue}",
+                               severity=Severity.ERROR,
+                               exception=exc).publish()
+            return None
+
+    def issues_opened(self):
+        return any(issue.state != 'closed' for issue in self.issues)
+
+    def issues_labeled(self):
+        branch_version = '.'.join(self.params.get('scylla_version').split('.')[0:2])
+        issues_labels = sum([issue.labels for issue in self.issues], [])
+
+        return any(f'sct-{branch_version}-skip' in label.name for label in issues_labels)
+
+    def __bool__(self):
```

Comment:
Conluding, if any issue is opened or it contains `sct-{branch_version}-skip` in label, then it resolves to `True`
Worth to add it to docstring so users will know how to add skip label to issue.

Reply:
I'll add it

Reply:
added a docstring to the class